src/components/tutor-dialog.cjsx

```diff
@@ -0,0 +1,63 @@
+BS = require 'react-bootstrap'
+React = require 'react'
+_ = require 'underscore'
+{Promise} = require 'es6-promise'
+
+TutorDialog = React.createClass
+
+  displayName: 'TutorDialog'
+
+  propTypes:
+    title:     React.PropTypes.string.isRequired
+    onProceed: React.PropTypes.func.isRequired
+    onCancel:  React.PropTypes.func.isRequired
+    body:      React.PropTypes.element.isRequired
+    show:      React.PropTypes.bool
+    buttons:   React.PropTypes.arrayOf( React.PropTypes.element )
+    className: React.PropTypes.string
+
+  getInitialState: ->
+    show: true
+
+  componentWillReceiveProps: (nextProps) ->
+    @setState(show: nextProps.show) if nextProps.show?
+
+  hide: ->
+    @setState(show: false)
+
+  render: ->
+    return null unless @state.show
+    buttons = @props.buttons or [
+        <BS.Button key='cancel' onClick={_.compose(@props.onCancel,  @hide)}>Cancel</BS.Button>
+        <BS.Button key='ok'     onClick={_.compose(@props.onProceed, @hide)} bsStyle='primary'>OK</BS.Button>
+    ]
+    classes = ['tutor-dialog']
+    classes.push @props.className if @props.className
+
+    <BS.Modal className={classes.join(' ')} onRequestHide={@hide} title={@props.title}>
+      <div className='modal-body'>
+        {@props.body}
+      </div>
+      <div className='modal-footer'>
+        {buttons}
+      </div>
+    </BS.Modal>
+
+module.exports = {
+
+  show: (props) ->
+    new Promise (onProceed, onCancel) =>
+
+      props = _.extend(_.clone(props), {
+        onProceed, onCancel, show: true
+      })
+
+      if @dialog
+        @dialog.replaceProps(props)
+      else
+        div = document.body.appendChild( document.createElement('div') )
+        @dialog = React.render(React.createElement(TutorDialog, props), div)
```

Comment:
It seemed to me that it'd be easiest if the Modal was callable from a component at any point in it's React lifecycle.  If we rendered the modal as a child of a component, it would have to track state and be more complex.  For instance the AccountLink would have to look something like:
```coffee
  redirectToAccount: ->
    @setState(askNavAllowed: true)

  render: ->
    if @state.askNavAllowed
       modal = <TutorDialog title="Can I nav?">
         {messages from somewhere}
       </TutorDialog>
    <BS.MenuItem onSelect={@redirectToAccount} >
      {modal}
      My Account
    </BS.MenuItem>
```

Whereas now it's just:
```coffee
  redirectToAccount: ->
    TransitionAssistant.checkTransitionStateTo("Account Profile Page").then ->
      window.location.href = CurrentUserStore.getProfileUrl()
```